draft-ietf-httpbis-header-structure.md

```diff
@@ -777,7 +781,8 @@ NOTE: This algorithm parses both Integers {{integer}} and Floats {{float}}, and
    6. If type is "float" and input_number contains more than 16 characters, fail parsing.
 8. If type is "integer":
    1. Parse input_number as an integer and let output_number be the product of the result and sign.
```

Comment:
I totally agree with Kazuho here. It's mandatory to check for overflows during the parsing, like we do here for example : http://git.haproxy.org/?p=haproxy.git;a=blob;f=src/h1.c;hb=HEAD#l52
Many integer parsers will reliably provide the information about the fact that an overflow was detected, but when it comes to floats it might not necessarily be the case (since very large numbers can still fit in a float), however the check against bounds is expected to work.

This means that we should mandate the use of a reliable parser known for detecting overflows, AND then check for the output number to fall within our bounds.

We should also warn against the risk of using generic parsers which as a byproduct of checking for errors, also happen to support base prefixes like "0x", or float parsers supporting exponents like "1e32" or special values like "inf", "-inf", "nan", all of which defeat length checks.

Reply:
> We should also warn against the risk of using generic parsers which as a byproduct of checking for errors, also happen to support base prefixes like "0x", or float parsers supporting exponents like "1e32" or special values like "inf", "-inf", "nan", all of which defeat length checks.

At this point in the algorithm the string has been preprocessed and is guaranteed to have no more than 19 characters, all of which are decimal digits. Or do you mean explicit warning against replacing the entire integer-parsing algorithm with a generic `to_int()` or equivalent?